docs/authoring.md

```diff
@@ -259,3 +224,112 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server from the AsyncAPI specification file you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. Use this parameter in case your template relies on users' information about what server from the specification file they want to use during generation. You also need this parameter if you want to use `server.protocol` notation within `conditionalFiles` configuration option. Once you decide to specify this parameter for your template, it is recommended you make it a mandatory parameter otherwise feature like `conditionalFiles` is not going to work if your users do not use this parameter obligatory.
+
+
+## React
+[React](https://reactjs.org) is the render engine which we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatibility.
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better error stack traces once something is not working.
+* Provides better support for separating code into more manageable chunks/components.
+* The readability of the template are much better then Nunjucks syntax.
+* Better tool support for development.
+
+When writing react templates you decide whether to use CommonJS, ES5 or ES6 modules since everything is bundles together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
+There you can find information about how the renderer works or how we transpile your template files.
+
+Your React template always require 3 dependencies `react`, `source-map-support` and `@asyncapi/generator-react-sdk`. The `react` dependency is required to declare React components inside the template. `source-map-support` ensure that the transpiled files are mapped to the actual template files, this is only to enable correct error stack trace. `@asyncapi/generator-react-sdk` is required to access the `File` component which is required as a root component for a file to be rendered. Furthermore it provides some common components to make your development easier, like `Text` or `Indent`.
```

Comment:
```suggestion
Your React template always requires 3 dependencies `react`, `source-map-support`, and `@asyncapi/generator-react-sdk`. The `react` dependency is required to declare React components inside the template. `source-map-support` ensures that the transpiled files are mapped to the actual template files, this is only to enable correct error stack trace. `@asyncapi/generator-react-sdk` is required to access the `File` component which is required as a root component for a file to be rendered. Furthermore, it provides some common components to make your development easier, like `Text` or `Indent`.
```

> The `react` dependency is required to declare React components inside the template.

Is it? Can't we inject it somehow with Rollup? For instance, when using Next.js it's not necessary. I wonder how they do it.

> `source-map-support` ensures that the transpiled files are mapped to the actual template files, this is only to enable correct error stack trace.

Can't we just include it as part of the `generator-react-sdk`? Just so people don't have to do it.

Reply:
🤔 It would be optimal, I'll try and see what I can do.

Reply:
I have added all the dependencies to the `generator-react-sdk` library. Everything works fine when its used as a normal dependency. However because we in development use a local dependency we are required to add these dependencies our self because npm installs them locally instead of "Merging" them with the template dependencies. 

Not sure how we should communicate this... Maybe I'll add it to the SDK readme? 🤔

Reply:
Not sure I follow 😅 What do you mean by "merging"? Why is it not working in development?

Reply:
At the moment in the templates we referer to the SDK with the following: `"@asyncapi/generator-react-sdk": "file:../generator-react-sdk"` i.e. a local dependency. However the way NPM treats this dependency is different from `"@asyncapi/generator-react-sdk": "0.1.0"`.

With a local dependency it installs IT's node_modules inside `../generator-react-sdk/node_modules` i.e. those dependencies are not installed inside `template/node_modules` as it normally would. This means that when the SDK has `source-map-support` dependency the template does not have access to this dependency if we use local dependencies.

Did that make sence in any way? 😅

Reply:
Oh! I see what you mean. Why is the `generator-react-sdk` a local dependency? Isn't it going to be another published package? I mean, can't we just replace `file:../generator-react-sdk` with `"@asyncapi/generator-react-sdk": "0.1.0"` and it would work?

If you want to work with a local dependency yourself while developing the `generator-react-sdk` package you can use [npm link](https://docs.npmjs.com/cli/v6/commands/npm-link).

Reply:
Yes thats what we should have done but didnt :D So it should fix itself :)